File: swap2num.cpp
#include<iostream>
using namespace std;

int main()
{
    int n1, n2, temp;

    cout <<"Enter 2 numbers:";
    cin >>n1 >>n2;


    //swap numbers using temp
  /*  temp = n1;
    n1 = n2;
    n2 = temp;*/

    //swap without using temp
    n2 = n1 + n2;   //total = n1 + n2
    n1 = n2 - n1;   //n1(n2 value) = total - n1
    n2 = n2 - n1;   //n2 (n1 value)= total - n1

    
    cout <<"n1: "<<n1 <<endl<<"n2: "<<n2<<endl;
    
    return 0;
}

File: Java/Problems/Swap2Num.java
public class Swap2Num {
    public static void main(String[] args) {
        
        int n1, n2, temp;

        n1 = 4;
        n2 = 5;

        //swap using temp
        temp = n1;
        n1 = n2;
        n2 = temp;

        System.out.println("n1 is " + n1);
        System.out.println("n2 is " + n2);
    }
}
